e2e/pvc.go

```diff
@@ -94,34 +94,38 @@ func deletePVCAndPV(c kubernetes.Interface, pvc *v1.PersistentVolumeClaim, pv *v
 
 	timeout := time.Duration(t) * time.Minute
 	start := time.Now()
+
+	pvcToDelete := pvc
 	err = wait.PollImmediate(poll, timeout, func() (bool, error) {
 		// Check that the PVC is deleted.
-		e2elog.Logf("waiting for PVC %s in state %s to be deleted (%d seconds elapsed)", pvc.Name, pvc.Status.String(), int(time.Since(start).Seconds()))
-		_, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
+		e2elog.Logf("waiting for PVC %s in state %s to be deleted (%d seconds elapsed)", pvcToDelete.Name, pvcToDelete.Status.String(), int(time.Since(start).Seconds()))
+		pvcToDelete, err = c.CoreV1().PersistentVolumeClaims(pvcToDelete.Namespace).Get(context.TODO(), pvcToDelete.Name, metav1.GetOptions{})
 		if err == nil {
 			return false, nil
 		}
 		if !apierrs.IsNotFound(err) {
-			return false, fmt.Errorf("get on deleted PVC %v failed with error other than \"not found\": %w", pvc.Name, err)
+			return false, fmt.Errorf("get on deleted PVC %v failed with error other than \"not found\": %w", pvcToDelete.Name, err)
```

Comment:
as we are doing get above at line 102, in case of the error the `pvcToDelete.Name` will be empty here?

Reply:
Yes, possibly. Or `pvcToDelete` might be `nil`. Either way, it will cause a failure. I opted for a potential panic, but have just reconsidered this and will use `pvc.Name` instead.